File: tp2_python/polygone.py
import math


def polygone():
    try:
        mesure_dun_cote = int(input("Entrez la mesure d'un côté"))
    except ValueError:
        print("vous avez entré une mauvaise valeur")
    try:
        nombre_de_cote = int(input("Entrez le nombre de côté"))
    except ValueError:
        print("vous avez entré une mauvaise valeur")
    aire_du_polygone = (nombre_de_cote * (mesure_dun_cote ** 2)) / (4 * math.tan(math.pi / nombre_de_cote))
    print(aire_du_polygone)

polygone()

File: BonusC++/Polygone.cpp
#include"Header.h"



	int Polygone() {
		int NombreDeCote;
		float MesureDunCote;
		double x, AireDuPolygone;
		const double M_PI = 3.14159265358979323846;
		
		cout << "Entrez la mesure des cotes " << endl;
		cin >> MesureDunCote;
		cout << "Entrez le nombre de cote" << endl;
		cin >> NombreDeCote;
		x = 180 * ((M_PI) /180);
		AireDuPolygone = (NombreDeCote * MesureDunCote * MesureDunCote) / (4 * tan((x/NombreDeCote)));
		cout << "L'aire de votre polygone est de " << AireDuPolygone << "m²" << endl;
		return 0;
	}
